tests/test_transitfit.py

```diff
@@ -0,0 +1,296 @@
+    )
+
+    assert is_converged
+    assert_allclose(tlsr['period'], 3.495, atol=1e-2)
+    assert_allclose(mafr['fitinfo']['finalparams']['period'], 3.495, atol=1e-2)
+
+    # theoretical t0 for this data (like 5 or 6 transits, over 1 TESS sector)
+    # is 3.99e+00 min = 6.64e-02 h = 2.77e-03 days.
+    assert mafr['fitinfo']['finalparamerrs']['std_perrs']['t0'] < 5e-3
+    assert mafr['fitinfo']['finalparamerrs']['std_merrs']['t0'] < 5e-3
+
+    # guess-timate period should be better than 3 minutes too.
+    assert mafr['fitinfo']['finalparamerrs']['std_perrs']['period'] < 3/(24*60)
+    assert mafr['fitinfo']['finalparamerrs']['std_merrs']['period'] < 3/(24*60)
+
+
+def test_fivetransitparam_fit_magseries_hard():
+    """
+    Fit one TESS sector of data, with a hard WJ (candidate) in it, for a
+    transit model (t0, period, incl, sma, rp/star) with believable error bars.
+    """
+
+    # path and identifier for GaiaDR2 4827527233363019776
+    lcpath = LCPATHS[1]
+    identifier = str(lcpath.split('gaiatwo')[1].split('-')[0].lstrip('0'))
+
+    hdul = fits.open(lcpath)
+    hdr, lc = hdul[0].header, hdul[1].data
+    hdul.close()
+
+    time = lc['TMID_BJD']
+    mag = lc['TFA2']
+    mag_0, f_0 = 12, 1e4
+    flux = f_0 * 10**( -0.4 * (mag - mag_0) )
+    flux /= np.nanmedian(flux)
+    err = np.ones_like(flux)*1e-4
+
+    teff = hdr['TICTEFF']
+    rstar = hdr['TICRAD']
+    logg = hdr['TICLOGG']
+
+    fit_savdir = os.path.join(os.getcwd(), 'fivetransitparam_results')
+    chain_savdir = os.path.join(os.getcwd(), 'fivetransitparam_chains')
+
+    # Autocorrelation time is like 400 steps for this case. But it does
+    # converge.
+    mafr, tlsr, is_converged = fivetransitparam_fit_magseries(
+                time, flux, err,
+                teff, rstar, logg,
+                identifier,
+                fit_savdir,
+                chain_savdir,
+                n_mcmc_steps=25000,
+                overwriteexistingsamples=False,
+                n_transit_durations=5,
+                make_tlsfit_plot=True,
+                exp_time_minutes=30,
+                bandpass='tess',
+                magsarefluxes=True,
+                nworkers=32
+    )
+
+    print(is_converged)
+    print(tlsr['period'])
+    print(mafr['fitinfo']['finalparams']['period'])
+
+    assert is_converged
+    assert_allclose(tlsr['period'], 10.714, atol=1e-2)
+    assert_allclose(mafr['fitinfo']['finalparams']['period'], 10.714, atol=1e-2)
+
+    # exofopTESS quotes t0 for TOI-450 to <1 minute (multisector). < 10 minutes
+    # required.
+    assert mafr['fitinfo']['finalparamerrs']['std_perrs']['t0'] < 10/(24*60)
+    assert mafr['fitinfo']['finalparamerrs']['std_merrs']['t0'] < 10/(24*60)
+
+    # guess-timate period should be better than say 10 minutes.
+    assert mafr['fitinfo']['finalparamerrs']['std_perrs']['period'] < 10/(24*60)
+    assert mafr['fitinfo']['finalparamerrs']['std_merrs']['period'] < 10/(24*60)
+
+
```

Comment:
Would be good to mark this test as xfail if it's mostly expected to fail, or mark it as skip if it's not going to work at all for now: http://doc.pytest.org/en/latest/skipping.html#skipping-test-functions. I'd go with skip for now because all the tests take an hour already.

Reply:
nice! good to know this exists.

Reply:
(done)